### PR TITLE
Add API29 full config for sargo

### PR DESCRIPTION
**Pull request by CaseyBakey**
Add API29 full config for sargo

### CODE REVIEW

**Comment by chirayudesai** [comment]
I'm not sure if we want to merge this at this point, especially given that it's only for one device, and includes Google apps.

Previous full configs included some bits from what you added but not google apps from what I can tell, since one can't actually ship them anyway.

**Comment by CaseyBakey** [comment]
As said in the README.md: "On the other hand the full configuration group (enabled with -f|--full flag from master script) has additional blobs & module targets which are normally marked as non-essential, although might be required for some carriers or in case of GApps being installed (either manually post-boot or included as additional vendor blobs)."

So the whole point of the "full" configurations was always to allow for Google Apps inclusion.

And I didn't add any new Google Apps: I based my "work" on the API-28 "full" configuration for the sargo device where you can find some Google Apps already included. It's just that some paths were moved between system/product/vendor partitions.

It's only for one device since this is the one I have and can test on right now. Other people will be able to do the same for other devices they own.

To be honest, I didn't expect to have my first PR ever, be rejected like that since this PR does what it says; add support of the README.md-detailed "full" configuration, for API-29, on sargo device. 

I did this because I need to be able to include Google Apps if the user wants to, in my https://github.com/CaseyBakey/chaosp project.

To be able to add Google Apps, thanks to opengapps/aosp_build project, we first need to have a working "full" configuration for the right device/API.

File: sargo/config.json [hunk 1]
```diff
@@ -527,6 +527,79 @@
       "BoardConfigVendor": [],
       "device-vendor": [],
       "vendor-skip-files": []
+    },
+    "full": {
+      "overlays-dir": "",
+      "system-bytecode": [
+        "system/app/atfwd/atfwd.apk",
+        "system/app/ConferenceDialer/ConferenceDialer.apk",
+        "system/priv-app/GoogleDocumentsUIPrebuilt/GoogleDocumentsUIPrebuilt.apk",
```
Comment (chirayudesai) [inline sargo/config.json]: This and a bunch of apps below are Google Apps - which is what I meant when I said we aren't allowed to ship Google Apps anyway.

File: sargo/config.json [hunk 2]
```diff
@@ -527,6 +527,79 @@
...
+        "system/priv-app/GooglePermissionControllerPrebuilt/GooglePermissionControllerPrebuilt.apk",
+        "product/app/ModuleMetadataGooglePrebuilt/ModuleMetadataGooglePrebuilt.apk",
+        "product/app/DiagnosticsToolPrebuilt/DiagnosticsToolPrebuilt.apk::PRESIGNED",
+        "product/app/VZWAPNLib/VZWAPNLib.apk",
+        "product/priv-app/AmbientSensePrebuilt/AmbientSensePrebuilt.apk::PRESIGNED",
+        "product/priv-app/AppDirectedSMSService/AppDirectedSMSService.apk",
+        "product/priv-app/ConnMO/ConnMO.apk",
+        "product/priv-app/DCMO/DCMO.apk",
+        "product/priv-app/DiagMon/DiagMon.apk",
+        "product/priv-app/EuiccGoogle/EuiccGoogle.apk::PRESIGNED",
```
Comment (chirayudesai) [inline sargo/config.json]: OTOH, this is something that depends on a Google app - namely, GMS, which is what the intention of the full config was previously.

**Review by chirayudesai — COMMENTED** [review]
> As said in the README.md: "On the other hand the full configuration group (enabled with -f|--full flag from master script) has additional blobs & module targets which are normally marked as non-essential, although might be required for some carriers or in case of GApps being installed (either manually post-boot or included as additional vendor blobs)."

Yes, the point was (from what I can tell since I'm not the original author) to add apps that may depend on Google Apps, not Google apps themselves.

> 
> So the whole point of the "full" configurations was always to allow for Google Apps inclusion.

But not directly include them here.
> 
> And I didn't add any new Google Apps: I based my "work" on the API-28 "full" configuration for the sargo device where you can find some Google Apps already included. It's just that some paths were moved between system/product/vendor partitions.

I don't see any of the `Google*` apps you added on https://github.com/AOSPAlliance/android-prepare-vendor/blob/android10/sargo/config.json (and even other devices, I did a quick check before commenting)
> 
> It's only for one device since this is the one I have and can test on right now. Other people will be able to do the same for other devices they own.

Makes sense. The same configuration can be copied for bonito too since they're identical (and really can share the same config too if we had the infrastructure for that.) 
> 
> To be honest, I didn't expect to have my first PR ever, be rejected like that since this PR does what it says; add support of the README.md-detailed "full" configuration, for API-29, on sargo device.

I apologize for that, I should not have closed the PR, simply left a comment. However, my other point still stands.
> 
> I did this because I need to be able to include Google Apps if the user wants to, in my https://github.com/CaseyBakey/chaosp project.
> 
> To be able to add Google Apps, thanks to opengapps/aosp_build project, we first need to have a working "full" configuration for the right device/API.

Ok, then we could add full config that's like what was present before, without adding any additional Google* apps.

**Comment by chirayudesai** [comment]
Also, we didn't add full configs here because none of the current users use it.

Happy to include proper full configs otherwise, if you want to add sargo could you add bonito too? Since they're the exact same device it'd be better to keep them in sync.

Others can pick to other devices later depending on what they can test.


Another note is that we're moving off config.json entirely - see https://github.com/AOSPAlliance/android-prepare-vendor/issues/15

We can still have the naked and full configs this way too, in fact the full config could just be the upstream lineage config (plus any minor additions if needed) since they do include most of this stuff.

I myself use a semi-full config but maintain it downstream instead (since we ship with microG but not full google apps, but can still use some of their apps)

**Review by chirayudesai — CHANGES_REQUESTED** [review]
Marking as changes requested.

File: sargo/config.json (hunk 1 above)
Comment (CaseyBakey) [reply sargo/config.json]: Yes, I did check again and that seem to be apps that I added to the config.json after received compilation (CHAOSP + opengapps/aosp_build) errors about them. Obviously GoogleDocumentsUIPrebuilt.apk shouldn't be included, you're right.
I can't determine what ConferenceDialer.apk is and it doesn't seem to be AOSP-based so yes, this one may also be discarded.
But the atfwd.apk comes from Qualcomm and is a blob binary needed for WiFi Display/Miracast to work so I think it belongs there.

Anyway, this project is not shipping anything, it just keeps a list of which binary blobs to extract from publicly available zip archive files, to a local AOSP build tree. It's nowhere near the legal problem some custom ROMs had in the past by directly including the Google Apps binaries in the already built custom ROM.

File: sargo/config.json (hunk 2 above)
Comment (CaseyBakey) [reply sargo/config.json]: Yes, I can only agree on that.

**Comment by CaseyBakey** [comment]
Yes, I can also add it to bonito if you say they're the same.

I will also be able to do it for blueline once I finish a successful CHAOSP build for the sargo device. I can also have access to this device.

I know that not a lot of people are using this project, and less are using the "full" configuration but it is still useful for some so I would be glad for it to stay/to be added since it doesn't interfere with the "naked" configurations.

I didn't know that you were moving from config.json files, I'll take a look on this issue.

But it seems you're teaching me a lot on the "Lineage" fact: I didn't know they were maintening such config. Do you have a link for me to check on it?

I'm also curious about you semi-full config.

I'm myself targeting "pico" config of opengapps in a first try and if everyting is working as intended on Android 10, move to microG too.

**Comment by CaseyBakey** [comment]
Ok the changes are done (but take a look at my answer here (https://github.com/AOSPAlliance/android-prepare-vendor/pull/17#discussion_r488047071).

I also added bonito but I have to trust you for this one :p

**Comment by chirayudesai** [comment]
> Yes, I can also add it to bonito if you say they're the same.

They are! If you compare the configs, you'll find that the only change is in the name.
> 
> I will also be able to do it for blueline once I finish a successful CHAOSP build for the sargo device. I can also have access to this device.

Do it for crosshatch too when you test blueline, same situation there.
> 
> I know that not a lot of people are using this project, and less are using the "full" configuration but it is still useful for some so I would be glad for it to stay/to be added since it doesn't interfere with the "naked" configurations.

For sure. If you're testing and using it, we're happy to accept your PRs.
> 
> I didn't know that you were moving from config.json files, I'll take a look on this issue.
> 
> But it seems you're teaching me a lot on the "Lineage" fact: I didn't know they were maintening such config. Do you have a link for me to check on it?

The issue has links to the lineage configs, and also all other details.
> 
> I'm also curious about you semi-full config.

https://gitlab.com/calyxos/vendor_android-prepare-vendor/-/commits/android10

eSIM support is one of the big changes right now, but I was going to add CarrierConfig and friends as needed later.
> 
> I'm myself targeting "pico" config of opengapps in a first try and if everyting is working as intended on Android 10, move to microG too.

**Comment by chirayudesai** [comment]
List seems ok, looks like it matches API 28, thanks!

Could you squash the fixup commits into the original? Ok to have one for sargo and one for bonito

**Comment by CaseyBakey** [comment]
So I didn't know how to squash commits...

I put each device changes in a branch, and I reset my master (--hard) to yours, then "git push -f" to my fork repo so it seems it auto-closed this PR.

I did create 2 PR, one for each device.

Now, I'll only work on branches :o

**Pull request closed without merging**